Code/Editor/ViewportTitleDlg.cpp

```diff
@@ -246,15 +246,33 @@ void CViewportTitleDlg::SetupHelpersButton()
             helperAction, &QAction::triggered, this,
             [this]
             {
-                m_ui->m_helpers->setChecked(AzToolsFramework::HelpersVisible() || AzToolsFramework::IconsVisible());
+                m_ui->m_helpers->setChecked(
+                    AzToolsFramework::HelpersVisible() || AzToolsFramework::IconsVisible() ||
+                    AzToolsFramework::OnlyShowHelpersForSelectedEntities());
             });
 
         auto iconAction = MainWindow::instance()->GetActionManager()->GetAction(AzToolsFramework::Icons);
         connect(
-            iconAction, &QAction::triggered, this,
+            iconAction,
+            &QAction::triggered,
+            this,
             [this]
             {
-                m_ui->m_helpers->setChecked(AzToolsFramework::HelpersVisible() || AzToolsFramework::IconsVisible());
+                m_ui->m_helpers->setChecked(
+                    AzToolsFramework::HelpersVisible() || AzToolsFramework::IconsVisible() ||
+                    AzToolsFramework::OnlyShowHelpersForSelectedEntities());
+            });
+
+        auto onlySelectedAction = MainWindow::instance()->GetActionManager()->GetAction(AzToolsFramework::OnlyShowHelpersForSelectedEntitiesAction);
+        connect(
+            onlySelectedAction,
+            &QAction::triggered,
+            this,
+            [this]
+            {
+                m_ui->m_helpers->setChecked(
+                    AzToolsFramework::HelpersVisible() || AzToolsFramework::IconsVisible() ||
+                    AzToolsFramework::OnlyShowHelpersForSelectedEntities());
```

Comment:
suggestion: Could we move these three calls to a little static helper at the top of the file, and then call that where all these three calls are, would just make the duplication a bit less and reduce the number of places we need to change if we add a new setting 🙂